google/ggrc-core
# GGRC-4864 Remove redundant workload on changing CycleTask status

# Issue description

On every POST/PUT on CycleTask all notifications are deleted for CycleTask. 
Then they are created back.
This leads to redundant workload on those operations.

# Steps to test the changes

- Create workflow with 100 cycletasks.
- Try to change 2 CycleTasks statuses quickly. Just change status of first and press to change status of the second one. Then repeat.
- Check that you don't have **'Lock timeout'** error in logs.

# Solution description

I've splitted notifications creation for POST and PUT on CycleTasks. Doing this I've removed redundant requests to DB.

# Sanity checklist




# PR Review checklist
